pkg/xray/provider_test.go

```diff
@@ -41,17 +43,50 @@ func getTestResty(t *testing.T) *resty.Client {
 	return restyClient
 }
 
-func testAccPreCheck(t *testing.T) {
+func testAccPreCheck(t *testing.T) error {
 	restyClient := getTestResty(t)
-	// TODO check the payload and make sure it's the right license type
-	_, err := restyClient.R().Get("/artifactory/api/system/licenses/")
+	err := checkArtifactoryLicense(restyClient)
 	if err != nil {
-		t.Fatal(err)
+		return err
 	}
 	ctx := context.Background()
 	provider, _ := testAccProviders["xray"]()
```

Comment:
I explained to alex a better way to do this

Reply:
I don't see how is it better, but I can implement it the same way.

Reply:
I have some problems with this, checkDestroy doesn't work

Reply:
You need to call the function with the key: https://github.com/jfrog/terraform-provider-project/blob/master/pkg/projects/provider_test.go#L40

Reply:
You need to call the func with a key: https://github.com/jfrog/terraform-provider-project/blob/b8e2e8a06bfaaf709bf0afecc6bce9109912765f/pkg/projects/util_test.go#L110

Reply:
Finally, it works. Thanks @alexhung!